libcxx/include/__config

```diff
@@ -1231,6 +1231,12 @@ typedef __char32_t char32_t;
 #    define _LIBCPP_HAS_EXPLICIT_THIS_PARAMETER 0
 #  endif
 
+#  if defined(_MSC_VER) || __SIZEOF_LONG_DOUBLE__ == __SIZEOF_DOUBLE__
```

Comment:
Is it guaranteed that if they have the same size they have the same representation?

Reply:
We're already using such condition in the test suite. I don't think we support any platform where their sizes are equal but their representations are not.

Reply:
I'm much less concerned about such a check in our test suite than in our ABI.

Reply:
I do agree it would be safer if the compiler provided some `#define` that told us basically "double and long double have the same size and representation". I don't know that this is provided by any compiler, though. One thing we could do is explicitly list the platforms on which we know that this is the case for certain. That way we'd be more conservative (i.e. the assumption may hold for some platforms that we wouldn't list, but at least we'd never make the assumptions for platforms where that's not true).

Reply:
The way to properly check that long double is the same as double is to use `LDBL_MANT_DIG == 53`.